test/unit/features/instance/methods-lifecycle.spec.js

```diff
@@ -53,6 +53,31 @@ describe('Instance methods lifecycle', () => {
         }
       }).$mount()
     })
+
+    it('Dep.target should be undefined during invocation of child immediate watcher', () => {
+      let calls = 0;
+      new Vue({
+        template: '<div><my-component></my-component></div>',
+        components: {
+          myComponent: {
+            template: '<div>hi</div>',
+            data() {
+              return { a: 123 }
+            },
+            watch: {
+              a: {
+                handler() {
+                  ++calls
+                  expect(Dep.target).toBe(undefined)
```

Comment:
Is this necessary?
I think I shouldn't have said unit test. A regression test closer to what you provided as a reproduction would be perfect

Reply:
Is this more like what you were thinking?
![image](https://user-images.githubusercontent.com/18545667/110326827-486fa900-806d-11eb-976a-e940d7123391.png)

Reply:
Yes, that looks like a better test for this scenario